lib/src/validator/flutter_constraint.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:pub_semver/pub_semver.dart';
+
+import '../entrypoint.dart';
+import '../validator.dart';
+
+/// Validates that a package's flutter constraint doesn't contain an upper bound
+class FlutterConstraintValidator extends Validator {
+  FlutterConstraintValidator(Entrypoint entrypoint) : super(entrypoint);
+  // TODO(sigurdm): Find a proper url for explaining this.
+  static const explanationUrl =
+      'https://github.com/flutter/flutter/issues/68143';
```

Comment:
Create a `dart.dev/go/flutter-upper-bound-deprecation` in:
https://github.com/dart-lang/site-www/blob/6e22397cad4de94c935290880dc273321b35489c/firebase.json#L135

I would suggest just linking to the issue, or something (anything), so that we just don't forget to create the link and redirect it somewhere :D

Reply:
Done